RetailCoder.VBE/Inspections/VariableRequiresSetAssignmentEvaluator.cs

```diff
@@ -0,0 +1,160 @@
+﻿using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Inspections
+{
+    class VariableRequiresSetAssignmentEvaluator
```

Comment:
I can definitely make it a static class (making `AllUserDeclarations` a parameter).  Regarding DeclarationFinder: If the DeclarationFinder 'finds declarations' then I would think rolling this functionality into it would be somewhat of an unexpected capability.  This class determines if a 'reference' meets conditions for a specific assignment protocol ('Set').  That said, I'm still sorting out the division(s) of responsibilities of everything in RD...so I can move it there...just let me know your preference.